### PR TITLE
Add concurrency init to api

### PR DESCRIPTION
**Pull request by ryanfkeepers**
expose middleware init via api layer

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :robot: Supportability/Tests

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/pkg/services/m365/api/client.go [hunk 1]
```diff
@@ -55,6 +56,14 @@ func NewClient(creds account.M365Config) (Client, error) {
 	return Client{creds, s, li, rqr}, nil
 }
 
+// initConcurrencyLimit ensures that the graph concurrency limiter is
+// initialized, so that calls do not step over graph api's service limits.
+// Limits are derived from the provided servie type.
+// Callers will need to call this func before making api calls an api client.
+func InitConcurrencyLimit(ctx context.Context, pst path.ServiceType) {
+	graph.InitializeConcurrencyLimiter(ctx, pst == path.ExchangeService, 4)
```
Comment (pandeyabs) [inline src/pkg/services/m365/api/client.go:64]: We should source parallelism from `control.Options` here?

Comment (ryanfkeepers) [reply src/pkg/services/m365/api/client.go:64]: No; the callers here don't provide control options at this time.  We can add that at a later date if needed.

Comment (pandeyabs) [reply src/pkg/services/m365/api/client.go:64]: works

File: src/pkg/services/m365/m365.go [hunk 2]
```diff
@@ -209,7 +210,7 @@ func GetUserInfo(
 	acct account.Account,
 	userID string,
 ) (*api.UserInfo, error) {
-	ac, err := makeAC(acct)
+	ac, err := makeAC(ctx, acct, path.ExchangeService)
```
Comment (pandeyabs) [inline src/pkg/services/m365/m365.go:213]: Sanity check: Is `path.ExchangeService` correct here?

**Review by pandeyabs — APPROVED** [review]

File: src/pkg/services/m365/m365.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/pkg/services/m365/m365.go:213]: `GetInfo` calls `GetMailbox` which works under exchange concurrency constraints.

**Pull request merged**